File: Cap14VB/ParserResultados.vb
Imports Cap14VB.Live.Search

Public Class ParserResultados
  Implements IEnumerable(Of Resultado)
  Private ReadOnly _results As IEnumerable(Of Result)

  Public Sub New(ByVal results As IEnumerable(Of Result))
    _results = results
  End Sub

  Public Function GetEnumeratorImpl() As IEnumerator(Of Resultado) Implements IEnumerable(Of Resultado).GetEnumerator
    Return GetEnumerator()
  End Function

  Public Function GetEnumerator() As IEnumerator(Of Resultado)
    Return _results.GetEnumerator()
  End Function

  Public Function GetEnumeratorImpl2() As IEnumerator Implements IEnumerable.GetEnumerator
    Return GetEnumerator()
  End Function
End Class

File: Cap14CSharp/ParserResultados.cs
using System.Collections;
using System.Collections.Generic;
using Cap14CSharp;
using Cap14CSharp.Live.Search;

namespace Cap14CSharp
{
    public class ParserResultados : IEnumerable<Resultado>
    {
        private readonly IEnumerable<Result> _results;

        public ParserResultados(IEnumerable<Result> results )
        {
            _results = results;
        }

        public IEnumerator<Resultado> GetEnumerator()
        {
            foreach (var item in _results)
            {
                yield return new Resultado{Descricao = item.Description , Titulo = item.Title, Url = item.Url};
            }
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            return GetEnumerator();
        }
    }
}